### PR TITLE
2.10.1 Changes

### PR DESCRIPTION
**Pull request by Salanto**
Patchnotes are :

**Major Changes**
Technical Preview of AO2-QtUI Files in Lobby/Serverlist (Partial #874)
 - Lobby now uses a new UI system to showcase the potential new UI System for future AO2 Releases.
 - Move Demoserver into its own category on the serverlist (#834)

**Minor Changes**
 - Add dialogue to add servers to favourites (#651)
 - Add Favorite Server Edit Dialog
 - Reworked Settings Dialogue to be resizable and Theme-Modifiable.
 - Add direct connection dialogue.
 - Sort themes dropdown according to Windows Explorer

**Deprecation**
 - Removed Case Announcements Feature and Network Handling
  - Settings options relating to case announcement will be removed from config.ini after the update
 - Remove migration code and checks for legacy serverlist.txt files
 - Remove Favorite Serverlist Context Menu

**Bugfixes**
 - Add logic to load a default soundfont for MIDI files. (base/soundfont.fs2 in the main mount path)
 - Fix character select screen not showing all characters until page changed or reloaded  (#854)
 - Restore evidence selector preview under Qt5 (#890) - Caused by #824 incorrectly overwriting signature.
 - Stretch speedlines by default (#895)
 - Change default audio behaviour to restore audio-settings from last launch (#903 )

### CODE REVIEW

**Comment by stonedDiscord** [comment]
that's a lot of damage

File: resource/ui/lobby_assets/down-arrow.png
Comment (stonedDiscord) [inline resource/ui/lobby_assets/down-arrow.png:1]: Have you considered using 🔽

File: resource/ui/lobby_assets/up-arrow.png
Comment (stonedDiscord) [inline resource/ui/lobby_assets/up-arrow.png:1]: Have you considered using ▲

**Review by stonedDiscord — COMMENTED** [review]
solid

File: src/charselect.cpp [hunk 1]
```diff
@@ -98,8 +98,14 @@ void Courtroom::set_char_select()
   set_size_and_pos(ui_char_taken, "char_taken");
   set_size_and_pos(ui_char_buttons, "char_buttons");
 
+  // Silence emission. This causes the signal to be emitted TWICE during server join!
+  // Fuck this. Performance Sandwich.
```
[the remark this answers is not in the document]
Comment (Salanto) [reply src/charselect.cpp:102]: Sandwiching those operations between the emission silencer reduces the amount of draws to 2, which is still suboptimal but better than nothing.

They are "sandwiched between." so to speak.

File: src/options.cpp [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+}
+
+void Options::setStopMusicOnCategoryEnabled(bool value)
+{
+  config.setValue("category_stop", value);
+}
+
+bool Options::logToTextFileEnabled() const
+{
+  return config.value("automatic_logging_enabled", true).toBool();
```
[the remark this answers is not in the document]
Comment (Salanto) [reply src/options.cpp:441]: What's so bad about it?

**Review by stonedDiscord — DISMISSED** [review]

**Review by stonedDiscord — DISMISSED** [review]

**Review by stonedDiscord — APPROVED** [review]

File: Attorney_Online.pro [hunk 3]
```diff
@@ -1,17 +1,19 @@
-QT += core gui widgets network websockets
+QT += core gui widgets network websockets uitools
```
[the remark this answers is not in the document]
Comment (Salanto) [reply Attorney_Online.pro:1]: Works according to Qt6 docs

**Review by Crystalwarrior — APPROVED** [review]
Cool and good!

**Pull request merged**